sql/codegen.go

```diff
@@ -276,59 +259,98 @@ classifier = {{.Estimator}}(
 	{{end}}
 
 {{if .Train}}
-def train_input_fn(features, labels, batch_size):
-    dataset = tf.data.Dataset.from_tensor_slices((dict(features), labels))
-    dataset = dataset.shuffle(1000).repeat().batch(batch_size)
-    return dataset
+def train_input_fn(batch_size):
+	feature_types = dict()
+	feature_shapes = dict()
+	for name in feature_column_names:
+		if column_name_to_type[name] == "categorical_column_with_identity":
+			feature_types[name] = tf.int64
+			feature_shapes[name] = tf.TensorShape([None])
+		else:
+			feature_types[name] = tf.float32
+			feature_shapes[name] = tf.TensorShape([])
+
+	gen = db_generator(driver, conn, """{{.StandardSelect}}""",
+		feature_column_names, "{{.Y.Name}}", column_name_to_type)
+	dataset = tf.data.Dataset.from_generator(gen, (feature_types, tf.int64), (feature_shapes, tf.TensorShape([1])))
+	dataset = dataset.shuffle(1000).repeat().batch(batch_size)
+	return dataset
 
 {{if .SelfDefined}}
 classifier.compile(optimizer=classifier.default_optimizer(),
 	loss=classifier.default_loss(),
 	metrics=["accuracy"])
-classifier.fit(train_input_fn(X, Y, BATCHSIZE),
+classifier.fit(train_input_fn(BATCHSIZE),
 	epochs=EPOCHS if EPOCHS else classifier.default_training_epochs(),
 	steps_per_epoch=STEPS, verbose=0)
 classifier.save_weights("{{.Save}}", save_format="h5")
 {{else}}
 classifier.train(
-    input_fn=lambda:train_input_fn(X, Y, BATCHSIZE),
-    steps=STEPS * (EPOCHS if EPOCHS else 1))
+	input_fn=lambda:train_input_fn(BATCHSIZE),
+	steps=STEPS * (EPOCHS if EPOCHS else 1))
 {{end}}
 
-def eval_input_fn(features, labels, batch_size):
-    dataset = tf.data.Dataset.from_tensor_slices((dict(features), labels))
-    dataset = dataset.batch(batch_size)
-    return dataset
+def eval_input_fn(batch_size):
```

Comment:
It looks like `eval_input_fn` and `train_input_fn` are similar.

Reply:
Done!